File: codificacao-c#/URI1024.cs
using System;
using System.Collections.Generic;

class URI
{
    public static string criptografarString(string str)
    {
        char[] ascii = str.ToCharArray();
        char[] asciiInvertido = new char[ascii.Length];
        for (int i = 0; i < ascii.Length; i++)
        {
            if (((int)ascii[i] <= 90 && (int)ascii[i] >= 65) || ((int)ascii[i] <= 122 && (int)ascii[i] >= 97))
            {
                ascii[i] = (char)((int)ascii[i] + 3);
            }
            asciiInvertido[ascii.Length - 1 - i] = ascii[i];
        }
        string strFinal = "";
        for (int i = 0; i < asciiInvertido.Length; i++)
        {
            if (i >= (asciiInvertido.Length / 2))
            {
                asciiInvertido[i] = (int)asciiInvertido[i] != 0 ? (char)((int)asciiInvertido[i] - 1) : (char)255;
            }
            strFinal += asciiInvertido[i];
        }
        return strFinal;
    }

    static void Main(string[] args)
    {
        int N = Convert.ToInt32(Console.ReadLine());
        List<string> lista = new List<string>();
        for (int i = 0; i < N; i++)
        {
            string str = Console.ReadLine();
            lista.Add(criptografarString(str));
        }
        foreach (string x in lista)
        {
            Console.WriteLine(x);
        }
    }
}

File: codificacao-java/Uri1024.java
import java.io.IOException;
import java.util.*;

public class Main {
	
	public static String criptografarString(String str) {
		char[] ascii = str.toCharArray();
		char[] asciiInvertido = new char[ascii.length];
		for(int i = 0; i < ascii.length; i++){
	        if(((int)ascii[i] <= 90 && (int)ascii[i] >= 65) || ((int)ascii[i] <= 122 && (int)ascii[i] >= 97)) {
	        	ascii[i] = (char)((int)ascii[i] + 3);
	        }
	        asciiInvertido[ascii.length - 1 - i] = ascii[i];
	    }
		String strFinal = "";
		for(int i = 0; i < asciiInvertido.length; i++) {
			if(i >= (asciiInvertido.length / 2)) {
				asciiInvertido[i] = (int)asciiInvertido[i] != 0 ? (char)((int)asciiInvertido[i] - 1) : (char)255;	
			}
	        strFinal += asciiInvertido[i];
		}
		return strFinal;
	}

	public static void main(String[] args) throws IOException {
        Scanner sc = new Scanner(System.in);
        String N = sc.nextLine();
    	List<String> lista = new ArrayList<>();
    	for(int i = 0; i < Integer.parseInt(N); i++) {
        	String str = sc.nextLine();
        	lista.add(criptografarString(str));
        }
    	lista.forEach(System.out::println);
	}

}